cms/djangoapps/contentstore/utils.py

```diff
@@ -354,6 +354,20 @@ def ancestor_has_staff_lock(xblock, parent_xblock=None):
     return parent_xblock.visible_to_staff_only
 
 
+def ancestor_has_optional_completion(xblock, parent_xblock=None):
+    """
+    Returns True if one of xblock's ancestors has optional completion.
```

Comment:
Unfortunately, this is not true. The following scenario is possible in Studio:
1. Mark a subsection as optional.
2. Mark the subsection as non-optional.
3. Mark one of the units as optional.
4. Mark the section as optional.

Now, you will see that the `ancestor_has_optional_completion` for units returns False. That's because the Inheritance mixin retrieves the `False` value from the subsection, i.e., when we change the value from `True` -> `False`, it no longer retrieves the value from its parent.

Reply:
Fixed in 836a127836600a813cd091c82a3db86766426137.

Reply:
oh, you're right! thanks for the fix :)